tools/sotn_build/get_file_list.py

```diff
@@ -0,0 +1,311 @@
+                path=assetPath, filename=seg[2], ext=extension.get(seg[1], seg[1])
+            )
+            for seg in segments
+            if len(seg) >= 3 and seg[1] in SEG_TYPES["asset"]
+        }
+        # Add a .bin file for each .dec file and supplement assetFiles with a directory listing
+        assetFiles.update(
+            {file.replace("dec", "bin") for file in assetFiles if ".dec" in file}
+        )
+    return assetFiles
+
+
+# Dedicated function for ParseArgs so that this file can be imported in a module and still have the ability to parse out the proper args
+def ParseArgs(args: list):
+    parser = argparse.ArgumentParser(
+        description="For gathering information in more complex ways than Make is meant to do"
+    )
+    parser.add_argument(
+        "--version",
+        required=False,
+        type=str,
+        default=os.getenv("VERSION") or "us",
+        help="Sets game version and overrides VERSION environment variable",
+    )
+    subparsers = parser.add_subparsers(dest="action", help="The action to perform")
+    parserList = subparsers.add_parser("list", help="Retrieve a list of files")
+    parserList.add_argument("ovl", type=str, help="Overlay to target")
+    parserList.add_argument(
+        "list_type",
+        type=str,
+        choices=("src_files", "o_files", "merged"),
+        help="What to list",
+    )
+    parserList.add_argument(
+        "-c",
+        "--config-type",
+        type=str,
+        choices=list(CONFIG_PATTERNS.keys()),
+        required=True,
+        help="Type of config to reference",
+    )
+    parserList.add_argument(
+        "-s",
+        "--include-shared",
+        action="store_true",
+        required=False,
+        help="Include shared src files with ovl src files",
+    )
+    parserList.add_argument(
+        "-H",
+        "--ignore-hidden",
+        action="store_true",
+        required=False,
+        help='Ignore file types that begin with "dot" in splat configs',
+    )
+    parserList.add_argument(
+        "-f",
+        "--asset-filter",
+        action="append",
+        type=str,
+        default="*",
+        required=False,
+        help="Filters to apply to asset_path files",
+    )
+
+    return parser.parse_args(args)
+
+
+# This logic is expected to be contained elsewhere if this file is imported as a module
+def main(args: list) -> None:
+    args = ParseArgs(args)
+    config = GetConfig(args)
+
+    assetFilters: tuple = tuple(
+        tuple(aFilter.split("*")) for aFilter in NormalizeArgs(args.asset_filter)
+    )
+    fileSegments: list = GetFileSegments(config["segments"])
+    asmFiles: set = GetAsmFiles(config["options"], fileSegments)
+    srcFiles: set = GetSrcFiles(config["options"], fileSegments)
+    assetFiles: set = GetAssetFiles(config["options"], fileSegments, assetFilters)
+
+    # These are currently sorted for ease of comparison/debugging, but it isn't relevant for actual function
```

Comment:
Just for performance reasons it might be worth removing this sorting before merging since it will have some small overhead (or perhaps make sorted output an option for debugging purposes).